### ISSUE TITLE
[CSS] Token after BOM can’t be `whitespace`, `at` or `EOF`

### ISSUE THREAD

**Issue by Jayman2000** [validator/validator#1302]
When a byte order mark is used, `.css` files often fail to validate. More specifically, there’s limitations as to which [tokens](https://www.w3.org/TR/css-syntax-3/#tokenization) the file can start with. For example, if the first token in the file is a space, then you’ll get a “Parse Error”. These same limitations don’t apply to files that don’t have BOMs.

[Comments aren’t tokens.](https://www.w3.org/TR/css-syntax-3/#consume-token) This (rather confusingly) means that if your `.css` file starts with a BOM followed by a comment followed by a newline, then it will fail to validate with a “Parse Error”. 

The following ZIP file has a bunch of examples that produce an error only when they have a BOM. It also contains some examples that don’t produce errors, regardless of whether or not they have a BOM.
~~[CSS validation problems.zip](https://github.com/validator/validator/files/7884975/CSS.validation.problems.zip)~~

Edit: Turns out one of the examples from that ZIP file (`Fails with BOM/7/with-bom.css`) did actually contain [invalid CSS](https://github.com/w3c/css-validator/issues/249#issuecomment-564320490). Here’s an updated version with that example removed:
[CSS validation problems v2.zip](https://github.com/validator/validator/files/7890525/CSS.validation.problems.v2.zip)

**Comment by nschonni** [comment]
Probably something upstream at https://github.com/w3c/css-validator

**Comment by Jayman2000** [comment]
> Probably something upstream at https://github.com/w3c/css-validator

I don’t think so. I tried all of the examples from the updated ZIP file with [the online version of that validator](https://jigsaw.w3.org/css-validator/), and they all passed.

**Comment by sideshowbarker** [comment]
> > Probably something upstream at [w3c/css-validator](https://github.com/w3c/css-validator)
> 
> I don’t think so. I tried all of the examples from the updated ZIP file with [the online version of that validator](https://jigsaw.w3.org/css-validator/), and they all passed.

Yes, I guess this failure is due to how our consuming code is invoking the stylesheet parser, which is this code:

https://github.com/validator/validator/blob/c17730732faf4985eae5a73ef89ce998e2f0136c/src/nu/validator/checker/schematronequiv/Assertions.java#L1561-L1564

I suppose I might need to add something to strip out any BOM from that `styleContents` string before parsing it.

**Comment by sideshowbarker** [comment]
Fixed and deployed to https://validator.w3.org/nu/
